crates/tx-parser/src/filter.rs

```diff
@@ -1,92 +1,94 @@
-    prelude::DepositTxParams,
+use strata_state::{
+    batch::SignedBatchCheckpoint,
+    tx::{DepositInfo, DepositRequestInfo, ProtocolOperation},
 };
-use strata_state::{batch::SignedBatchCheckpoint, tx::ProtocolOperation};
 
 use super::messages::ProtocolOpTxRef;
+pub use crate::filter_types::TxFilterConfig;
 use crate::{
     deposit::{deposit_request::extract_deposit_request_info, deposit_tx::extract_deposit_info},
     inscription::parse_inscription_data,
-    utils::generate_taproot_address,
 };
 
-/// kind of transactions can be relevant for rollup node to filter
-#[derive(Clone, Debug, PartialEq, BorshSerialize, BorshDeserialize)]
-pub enum TxFilterRule {
-    /// Inscription transactions with given Rollup name. This will be parsed by
-    /// InscriptionParser which dictates the structure of inscription.
-    RollupInscription(RollupName),
-    /// Deposit Request transaction
-    DepositRequest(DepositTxParams),
-    /// Deposit transaction with deposit config and address
-    Deposit(DepositTxParams),
-}
-
-type RollupName = String;
-
-/// Reads the operator wallet public keys from Rollup params. Returns None if
-/// not yet bootstrapped
-/// FIXME: This is only for devnet as these pks have to be read from the chain state
-fn get_operator_wallet_pks(params: &RollupParams) -> Vec<Buf32> {
-    let OperatorConfig::Static(operator_table) = &params.operator_config;
-
-    operator_table.iter().map(|op| *op.wallet_pk()).collect()
-}
-
-pub fn derive_tx_filter_rules(params: &RollupParams) -> anyhow::Result<Vec<TxFilterRule>> {
-    let operator_wallet_pks = get_operator_wallet_pks(params);
-    let address = generate_taproot_address(&operator_wallet_pks, params.network)?;
-    let deposit_provider = params.get_deposit_params(address);
-    Ok(vec![
-        TxFilterRule::RollupInscription(params.rollup_name.clone()),
-        TxFilterRule::DepositRequest(deposit_provider.clone()),
-        TxFilterRule::Deposit(deposit_provider),
-    ])
-}
-/// Filter all the relevant [`Transaction`]s in a block based on given [`TxFilterRule`]s
-pub fn filter_relevant_txs(block: &Block, filters: &[TxFilterRule]) -> Vec<ProtocolOpTxRef> {
+/// Filter protocol operations as refs from relevant [`Transaction`]s in a block based on given
+/// [`TxFilterConfig`]s
+pub fn filter_protocol_op_tx_refs(
+    block: &Block,
+    filter_config: TxFilterConfig,
+) -> Vec<ProtocolOpTxRef> {
     block
         .txdata
         .iter()
         .enumerate()
         .filter_map(|(i, tx)| {
-            check_and_extract_relevant_info(tx, filters)
+            extract_protocol_op(tx, &filter_config)
                 .map(|relevant_tx| ProtocolOpTxRef::new(i as u32, relevant_tx))
         })
         .collect()
 }
 
 ///  if a [`Transaction`] is relevant based on given [`RelevantTxType`]s then we extract relevant
-///  info
-fn check_and_extract_relevant_info(
+///  info.
+///  TODO: make this function return multiple ops as a single tx can have multiple outpoints that's
```

Comment:
Not part of docstring:
```suggestion
// TODO: make this function return multiple ops as a single tx can have multiple outpoints that's
```